core/audits/dobetterweb/dom-size.js

```diff
@@ -69,13 +71,47 @@ class DOMSize extends Audit {
     };
   }
 
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @param {LH.Audit.Context} context
+   * @return {Promise<number|undefined>}
+   */
+  static async computeTbtImpact(artifacts, context) {
+    let tbtImpact = 0;
+
+    // We still want to surface this audit in snapshot mode, but since we don't compute TBT
+    // the impact should always be undefined.
+    const {GatherContext, devtoolsLogs, traces} = artifacts;
+    if (GatherContext.gatherMode !== 'navigation') {
+      return undefined;
+    }
+
+    // Since the artifacts are optional, it's still possible for them to be missing in navigation mode.
+    // Navigation mode does compute TBT so we should surface a numerical savings of 0.
+    if (!devtoolsLogs?.[Audit.DEFAULT_PASS] || !traces?.[Audit.DEFAULT_PASS]) {
+      return 0;
+    }
+
+    const metricComputationData = Audit.makeMetricComputationDataInput(artifacts, context);
+
+    try {
+      const tbtImpactTasks = await TBTImpactTasks.request(metricComputationData, context);
+      for (const task of tbtImpactTasks) {
+        if (task.group.id !== 'styleLayout') continue;
+        tbtImpact += task.selfTbtImpact;
```

Comment:
This would benefit from some comments.

is the idea that the size / depth / whatever of the dom contributes to TBT only as much as style layout tasks do?

This opp can only be zero if you have no doms to style. is that the intention?

Reply:
> is the idea that the size / depth / whatever of the dom contributes to TBT only as much as style layout tasks do

Pretty much. This calculation assumes TBT impact of style/layout stuff is correlated to the overall DOM size.

> This opp can only be zero if you have no doms to style. is that the intention?

Every page spends a nonzero amount of main thread time on style/layout but not every page will block the main thread because of it. `selfTbtImpact` captures the blocking time not duration, so any layout/styles stuff that is not part of a long task will also have 0 impact here.

Reply:
Imagine a page that is otherwise scoring a perfect 100, with a large dom. But now add a JS setTimeout that does constant layout thrashing (force style recalcs) - will this result in a `dom-size` audit showing 100% of TBT being attributed as "reduce your dom size to fix TBT" - when really the problem is in the style recalcs?

Reply:
> will this result in a dom-size audit showing 100% of TBT being attributed as "reduce your dom size to fix TBT" - when really the problem is in the style recalcs?

Yes, but I have some thoughts:
- If the page has a large DOM and it forces style recalcs, then the large DOM is partially to blame for any blocking time IMO
- We could add a note in this audit's docs like "if a large DOM is necessary, try to minimize forced style recalcs"
- I'm not sure we can get a better estimate. Even if we had JS profiling data and knew which style/layout tasks were caused by manual style recalcs, my first point still forces us to consider them as impactful for this audit.